src/Camera/Camera.cpp

```diff
@@ -146,10 +143,10 @@ void Camera::DeprojectScreenToWorld(glm::vec2 screenCoord, glm::vec3& outWorldOr
 	const float screenSpaceY = ((1.0f - screenCoord.y) - 0.5f) * 2.0f;
 
 	// The start of the ray trace is defined to be at mousex,mousey,1 in
-	// projection space (z=0 is near, z=1 is far - this gives us better
+	// projection space (z=1 is near, z=0 is far - this gives us better
 	// precision) To get the direction of the ray trace we need to use any z
 	// between the near and the far plane, so let's use (mousex, mousey, 0.5)
-	const glm::vec4 rayStartProjectionSpace = glm::vec4(screenSpaceX, screenSpaceY, 0.0f, 1.0f);
+	const glm::vec4 rayStartProjectionSpace = glm::vec4(screenSpaceX, screenSpaceY, 0.9f, 1.0f);
 	const glm::vec4 rayEndProjectionSpace = glm::vec4(screenSpaceX, screenSpaceY, 0.5f, 1.0f);
```

Comment:
I have a hard time understanding this code and why 1.0f doesn't work.
Is there anything to do with the end value? Maybe you have to be in the negatives?

Reply:
Test pass at a value of 0.999 and right now i had test pass at 0.9995 so it seems we can approach one but if we are too close it starts to fail, maybe there is something i miss that mean we can't really reach 1 or it is a precision issue.

Reply:
I did some test selecting and teleporting them and it seems fine, i have some trouble selecting the villager but i had problem with that before, moving by fly-by seems also fine even with a start ray at z =1.f. In game having z=0.999f doesn't seems to change much but it passes the unit test and 1. don't.

Reply:
Could be that you're running into a Vanilla bug that we've enshrined in a unit test unfortunately

Reply:
I tried to get value for `DeprojectScreenToWorld` on frame 1 (fail at the start of frame 2) of MoveBackwardForward scenario while not reversing the perspective. This is when updating `_screenSpaceMouseRaycastHit`

Z = 1.f (Fail)
[998.717773 2.04257107 1118.00208] Origin
[997.435729 1.08520007 1116.00415] End
[-0.500863254 -0.37402162 -0.780540705] Direction

Z = 0.999f (Pass)
[998.716491 2.04161239 1118.00012] Origin
[997.435729 1.08520007 1116.00415] End
[-0.500856638 -0.374016047 -0.780547678] Direction

Normal Z = 0.f (Pass)
[998.717956 2.04264212 1118.0022] Origin
[997.436096 1.08534205 1116.00439] End
[-0.500833511 -0.374025106 -0.780558168] Direction